handlers.py

```diff
@@ -114,35 +114,54 @@ def newNamespace(spec, name, meta, logger, **kwargs):
 
 
-# Reload Pod when update configmap or secret
+# Reload deployment when update configmap or secret
 
 @kopf.on.update('', 'v1', 'configmaps', when=watch_namespace)
-def reload_pod_config(body, meta, spec, status, old, new, diff, **kwargs):
-    # Get namespace
-    ns = meta.namespace
-    api = kubernetes.client.CoreV1Api()
-    pods = api.list_namespaced_pod(ns)
-    print(ns, meta.name)
-    for pod in pods.items:
-        # Find which pods use this secrets
-        if pod.metadata.annotations and pod.metadata.annotations.get('synator/reload'):
-            if any('configmap:' + meta.name in s for s in pod.metadata.annotations.get('synator/reload').split(',')):
-                # Reload pod
-                api.delete_namespaced_pod(
-                    pod.metadata.name, pod.metadata.namespace)
-
+def reload_deployment_config(body, meta, spec, status, old, new, diff, logger, **kwargs):
+    reload_deployments_sync(meta, 'configmap', logger)
 
 @kopf.on.update('', 'v1', 'secrets', when=watch_namespace)
-def reload_pod_secret(body, meta, spec, status, old, new, diff, **kwargs):
-    # Get namespace
-    ns = meta.namespace
-    api = kubernetes.client.CoreV1Api()
-    pods = api.list_namespaced_pod(ns)
-    print(ns, meta.name)
-    for pod in pods.items:
-        # Find which pods use this secrets
-        if pod.metadata.annotations and pod.metadata.annotations.get('synator/reload'):
-            if any('secret:' + meta.name in s for s in pod.metadata.annotations.get('synator/reload').split(',')):
-                # Reload pod
-                api.delete_namespaced_pod(
-                    pod.metadata.name, pod.metadata.namespace)
+def reload_deployment_secret(body, meta, spec, status, old, new, diff, logger, **kwargs):
+    reload_deployments_sync(meta, 'secret', logger)
+
+def reload_deployments_sync(meta, secretOrConfigmap, logger):
+    try:
+        # Get namespace
+        ns = meta.namespace
+        api = kubernetes.client.AppsV1Api()
+        deployments = api.list_namespaced_deployment(ns)
+        configSearch = secretOrConfigmap + ':' + meta.name
+
+        logger.info(f"NS: %s Name: %s Deployments %s", ns, configSearch, str(len(deployments.items)))
+
+        for deployment in deployments.items:
+            annotations = deployment.spec.template.metadata.annotations
+            syncReloads = []
+            if annotations and annotations.get('synator/reload'):
+                syncReloads = annotations.get('synator/reload').split(',')
+            
+            if any(configSearch in s for s in syncReloads):
+                # Reload deployment
+                update_deployment(api, deployment, logger)
+    except kubernetes.client.rest.ApiException as e:
+        print("Exception when calling AppsV1Api: %s\n" % e)
+
+def update_deployment(api, deployment, logger):
+    # Update revision
+    revision = deployment.spec.template.metadata.annotations.get('deployment.neoassist/revision')
```

Comment:
Could you change  `deployment.neoassist/revision` to `synator/revision`

Reply:
Yes, I will make the adjustment